src/ConnectedMode/UI/ConnectedModeServices.cs

```diff
@@ -18,15 +18,37 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System.ComponentModel.Composition;
 using SonarLint.VisualStudio.ConnectedMode.Shared;
 using SonarLint.VisualStudio.Core;
 using SonarLint.VisualStudio.Core.Binding;
 
 namespace SonarLint.VisualStudio.ConnectedMode.UI;
 
-public record ConnectedModeServices(
-    IBrowserService BrowserService,
-    IThreadHandling ThreadHandling,
-    ISlCoreConnectionAdapter SlCoreConnectionAdapter,
-    IConfigurationProvider ConfigurationProvider,
-    ISharedBindingConfigProvider SharedBindingConfigProvider);
+public interface IConnectedModeServices
+{
+    public ISharedBindingConfigProvider SharedBindingConfigProvider { get; } 
+    public IBrowserService BrowserService { get; } 
+    public IThreadHandling ThreadHandling { get; } 
+    public ILogger Logger { get; } 
+    public ISlCoreConnectionAdapter SlCoreConnectionAdapter { get; } 
+    public IConfigurationProvider ConfigurationProvider { get; } 
+}
+
+[Export(typeof(IConnectedModeServices))]
+[method: ImportingConstructor]
+public class ConnectedModeServices(
```

Comment:
Add PartCreationPolicy attribute and the Mef tests

Reply:
It effectively is. You only import it once. Plus, it's immutable and I think all of the property types are singletons as well. But anyway, even if a class is not single, it's still better to explicitly add PartCreationPolicy (there's NonShared value of the enum)

Reply:
I can make it a singleton. I have seen classes that do not declare the PartCreationPolicy at all, so I was following that style.

Reply:
We can discuss that, but IMO it's wrong. We can also add that to the xtranet page afterwards. I will create a trello card for this

Reply:
I made it a singleton